### PR TITLE
[SPARK-6296] [SQL] Added equals to Column

### PR DESCRIPTION
**Pull request by vlyubin**

### CODE REVIEW

**Comment by sryza** [comment]
Mind including [SQL] in the title so that this gets sorted properly?

**Comment by rxin** [comment]
I think once you do that, you'd need to define hashCode as well?

File: sql/core/src/main/scala/org/apache/spark/sql/Column.scala [hunk 1]
```diff
@@ -59,6 +59,13 @@ class Column(protected[sql] val expr: Expression) {
 
   override def toString: String = expr.prettyString
 
+  override def equals(that: Any) = that match {
+    case that: Column => that.expr.equals(this.expr)
+    case _ => false
+  }
+
+  override def hashCode: Int = this.toString.hashCode
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Column.scala]: doesn't expr have its own hash code?

Comment (vlyubin) [inline sql/core/src/main/scala/org/apache/spark/sql/Column.scala]: I does. Switched to use that.

**Comment by rxin** [comment]
lgtm

**Comment by zsxwing** [comment]
I'm wondering if the meaning of `equals` in Column is precise, considering `Expression` does not implement `equals`.

**Comment by rxin** [comment]
As far as I know all concrete expressions are case classes, which should be fine.

**Pull request closed without merging**